File: Java/FilaEstatica/FilaEstatica/src/Enfileiravel.java
public interface Enfileiravel <T> {
    // métodos 1
    void enfileirar(T dado);
    T desenfileirar();
    T topo();
    void atualizar(T dado);

    // métodos 2
    boolean estaCheia();
    boolean estaVazia();
    String imprimir();
}

File: Kotlin/FilaEstatica/FilaEstatica/src/Enfileiravel.kt
interface Enfileiravel {
    // metodos 1
    fun enfileirar(dado: Any?)
    fun desenfilerar(): Any?
    fun frente(): Any?
    fun atualizar(dado: Any?)

    // metodos 2
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}